src/Extension.Tests/ChallengeTests.cs

```diff
@@ -43,10 +38,7 @@ public async Task teacher_can_access_code_from_submission_history_when_evaluatin
         {
             var capturedCode = new List<string>();
             var lesson = new Lesson();
-            using var kernel = new CompositeKernel
-            {
-                new CSharpKernel()
-            }.UseLessonEvaluateMiddleware(lesson);
+            using var kernel = CreateKernel(lesson);
             var challenge = GetEmptyChallenge();
             await lesson.StartChallengeAsync(challenge);
```

Comment:
the challenges should be added to the lesson before starting them, this code should throw exception, right?